modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/ActivationKeysTable/components/Header/hooks/useFilters.js

```diff
@@ -117,6 +117,17 @@ export default function useFilters(setFilterTerm, productName) {
 						}complimentary eq false`;
 					}
 
+					if (
+						environmentType === 'Production' &&
+						!filters.environmentTypes.value.includes(
+							'Non-Production'
+						)
+					) {
+						return `${accumulatorEnvironmentTypesFilter}${
+							index > 0 ? ' or ' : ''
+						}contains(productName, '${environmentType}') and not contains(productName, 'Non-Production')`;
+					}
```

Comment:
Note: @jairmedeiros this was a tricky suggested by Amos to unblock the issue with filter

This should be taken out someday